std/algorithm/iteration.d

```diff
@@ -11,11 +11,21 @@ $(T2 cache,
         Eagerly evaluates and caches another range's $(D front).)
 $(T2 cacheBidirectional,
         As above, but also provides $(D back) and $(D popBack).)
+$(T2 cartesianPower,
+        Lazily computes the Cartesian product of $(D r) for a number of
+        repetitions $(D repeat).
+        $(D"AB".cartesianPower(2)) returns $(D["AA", "AB", "BA", "BB"]))
 $(T2 chunkBy,
         $(D chunkBy!((a,b) => a[1] == b[1])([[1, 1], [1, 2], [2, 2], [2, 1]]))
         returns a range containing 3 subranges: the first with just
         $(D [1, 1]); the second with the elements $(D [1, 2]) and $(D [2, 2]);
         and the third with just $(D [2, 1]).)
+$(T2 combinations,
+        Lazily computes all k-combinations of $(D r).
+        $(D"AB".combinations(2)) returns $(D["AB"]))
+$(T2 combinationsRepeat,
```

Comment:
Hmm "arrangement" is defined

> In general, an arrangement of objects is simply a grouping of them. The number of "arrangements" of n items is given either by a combination (order is ignored) or permutation (order is significant).

(from: http://mathworld.wolfram.com/Arrangement.html)

This is really about allowing repeats

> the number of ways to sample k elements from a set of n elements allowing for duplicates (i.e., with replacement) but disregarding different orderings (e.g. {2,1,2} = {1,2,2}).

See also:

https://en.wikipedia.org/wiki/Combination#Number_of_combinations_with_repetition
https://en.wikipedia.org/wiki/Multiset#Counting_multisets